common/lib/xmodule/xmodule/tests/test_randomize_module.py

```diff
@@ -1,45 +1,104 @@
+                start="{start}">
+            <chapter url="ch1" url_name="chapter1" display_name="CH1">
                 <randomize url_name="my_randomize">
                 <html url_name="a" display_name="A">Two houses, ...</html>
                 <html url_name="b" display_name="B">Three houses, ...</html>
                 </randomize>
             </chapter>
+            <chapter url="ch2" url_name="chapter2" display_name="CH2">
+            </chapter>
          </course>
          '''.format(org=ORG, course=COURSE, start=start)
 
-        return system.process_xml(start_xml)
+        return self.system.process_xml(self.start_xml)
 
     def test_import(self):
         """
         Just make sure descriptor loads without error
         """
-        descriptor = self.get_dummy_course(START)
+        self.get_dummy_course(START)
+
+    def test_has_started(self):
+        """
+        Test CourseDescriptor.has_started.
+        """
+        self.course.start = _LAST_WEEK
+        self.assertTrue(self.course.has_started())
+        self.course.start = _NEXT_WEEK
+        self.assertFalse(self.course.has_started())
+
+    def test_children(self):
+        """ Check course/randomize module works fine """
+
+        self.assertTrue(self.course.has_children)
+        self.assertEquals(len(self.course.get_children()), 2)
+
+        def inner_get_module(descriptor):
+            """
+            Override systems.get_module
+                This method will be called when any call is made to self.system.get_module
+            """
+            if isinstance(descriptor, BlockUsageLocator):
+                location = descriptor
+                descriptor = self.modulestore.get_item(location, depth=None)
+            descriptor.xmodule_runtime = self.get_dummy_course()
+            descriptor.xmodule_runtime.descriptor_runtime = descriptor._runtime  # pylint: disable=protected-access
+            descriptor.xmodule_runtime.get_module = inner_get_module
+            return descriptor
+
+        self.system.get_module = inner_get_module
+
+        # Get randomize_descriptor from the course & verify its children
+        randomize_descriptor = inner_get_module(self.course.id.make_usage_key('randomize', 'my_randomize'))
+        self.assertTrue(randomize_descriptor.has_children)
+        self.assertEquals(len(randomize_descriptor.get_children()), 2)
 
-    # TODO: add tests that create a module and check.  Passing state is a good way to
-    # check that child access works...
+        # Call RandomizeModule which will select an element from the list of available items
+        randomize_module = RandomizeModule(
+            randomize_descriptor,
+            self.system,
+            DictFieldData({}),
+            ScopeIds(None, None, self.course.id, self.course.id)
+        )
+        self.assertTrue(randomize_module.child.display_name in ['A', 'B'])
+        self.assertTrue(randomize_module.get_children, 1)
```

Comment:
Its testing and verifying that the randomize module has two children and their `display_name`s are as expected.